templates/zerver/help/include/rest-endpoints.md

```diff
@@ -1,65 +1,65 @@
-* [Send a message](/api/send-message)
-* [Update a message](/api/update-message)
-* [Add an emoji reaction](/api/add-emoji-reaction)
-* [Remove an emoji reaction](/api/remove-emoji-reaction)
-* [Render a message](/api/render-message)
-* [Upload a file](/api/upload-file)
-* [Delete a message](/api/delete-message)
+* [Get messages](/api/get_messages)
+* [Get a raw message](/api/get_raw_message)
+* [Send a message](/api/send_message)
+* [Update a message](/api/update_message)
+* [Add an emoji reaction](/api/add_reaction)
+* [Remove an emoji reaction](/api/remove_reaction)
+* [Render a message](/api/render_message)
+* [Upload a file](/api/upload_file)
+* [Delete a message](/api/delete_message)
 * [Construct a narrow](/api/construct-narrow)
-* [Get a message's edit history](/api/get-message-history)
-* [Update a message's flags](/api/update-message-flags)
-* [Mark messages as read in bulk](/api/mark-as-read-bulk)
+* [Get a message's edit history](/api/get_message_history)
+* [Update a message's flags](/api/update_message_flags)
+* [Mark messages as read in bulk](/api/mark_all_as_read)
 
 #### Streams
 
-* [Get all streams](/api/get-all-streams)
-* [Get subscribed streams](/api/get-subscribed-streams)
-* [Add subscriptions](/api/add-subscriptions)
-* [Update subscription settings](/api/update-subscription-properties)
-* [Remove subscriptions](/api/remove-subscriptions)
-* [Get topics in a stream](/api/get-stream-topics)
-* [Topic muting](/api/mute-topics)
+* [Get all streams](/api/get_streams)
+* [Get subscribed streams](/api/list_subscriptions)
+* [Add subscriptions](/api/subscribe)
+* [Update subscription settings](/api/update_subscription_settings)
+* [Remove subscriptions](/api/unsubscribe)
+* [Get topics in a stream](/api/get_stream_topics)
+* [Topic muting](/api/mute_topic)
 * [Create a stream](/api/create-stream)
-* [Get stream ID](/api/get-stream-id)
-* [Delete a stream](/api/delete-stream)
-* [Update a stream](/api/update-stream)
+* [Get stream ID](/api/get_stream_id)
+* [Delete a stream](/api/delete_stream)
+* [Update a stream](/api/update_stream)
 
 #### Users
 
-* [Get all users](/api/get-all-users)
-* [Get a user](/api/get-user)
-* [Deactivate a user](/api/deactivate-user)
-* [Reactivate a user](/api/reactivate-user)
-* [Update a user](/api/update-user)
-* [Get profile](/api/get-profile)
-* [Create a user](/api/create-user)
-* [Set "typing" status](/api/typing)
-* [Get user presence](/api/get-presence)
-* [Get all user groups](/api/get-user-groups)
-* [Update notification settings](/api/update-notification-settings)
-* [Create a user group](/api/create-user-group)
-* [Update a user group](/api/update-user-group)
-* [Delete a user group](/api/delete-user-group)
+* [Get all users](/api/get_users)
+* [Get a user](/api/get_user_by_id)
+* [Deactivate a user](/api/deactivate_user_by_id)
+* [Reactivate a user](/api/reactivate_user)
+* [Update a user](/api/update_user)
+* [Get profile](/api/get_profile)
```

Comment:
This naming is kind of odd; it sounds like it's just like "get a user" but maybe is going to provide a different set of data about them.

But IIUC what it *actually* does is mostly that it gets data about yourself as a user. And then it has a couple of extra fields that maybe you aren't allowed to see about other users (or at least one, `pointer`), plus it's sort of a randomly different set of fields in general because it's evolved separately.

A better name might be `get_own_profile`, so it mentions the most distinctive feature.

Then I still don't love "profile" here because it's not really any more of a "profile" than the result of "Get a user" is. But the remaining confusion there may be mostly inherent in the actual API content, and not really about the name, once "[your] own" is mentioned.

Reply:
Or maybe `get_own_user()`?

Reply:
Yeah, `get_own_user` may be best.

It's still kind of confusing because it sounds like it should return the same thing as `get_user`, just as if you passed your own user ID there -- and it *almost* does, with just a couple of quirks. But it's not like "profile" really usefully communicates that distinction anyway.

Is there a long-term reason why this endpoint should be different from just passing your own user ID to `get_user`? The one thing I spotted in the example response that clearly doesn't belong in the latter is `pointer`, which IIUC is going away. If there's no long-term reason for a distinction here, and the differences are all either going away (`pointer`) or just quirks from separate evolution, then that's especially good for `get_own_user` being the name.

Reply:
Changed it to `get_own_user` for now.